msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerBaseStrategy.java

```diff
@@ -122,12 +126,16 @@ static AcquireTokenResult getAcquireTokenResult(@NonNull final Bundle resultBund
                     resultAdapter.authenticationResultFromBundle(resultBundle)
             );
 
+            Telemetry.emit(new BrokerEndEvent().isSuccessful(true));
```

Comment:
nit: I'd prefer calling "successful" event in the very same function that start() is called. (All entry and exit points of a telemetry event should be in the same function).

It might be redundant, but I think it's better in terms of readability/maintainability. (I know for sure by reading the function that I'm not missing any exit points).

Reply:
Sorry, I think I do not understand here. The isSuccessful() is to represent if the call succeeded, not the emit() of event.

Reply:
What I meant was that, if possible, every BrokerEndEvent() should be emitted in the same function as BrokerStartEvent().

Reply:
No, it should not. The start event and the end event do not necessary in the same function. This is why we need an atomic function like Telemetry.emit(). Please refer to the usage of UiStartEvent and UiEndEvent.

Reply:
hence 'if possible'.

The reason I suggested putting them in the same function is to avoid missing 'exit points' when we're making code change. Having them in one place, it's easier to spot if we forgot to emit any event when we're returning a value, or throw an exception. Putting them in multiple place, it is more likely that we might miss that (which will be very bad if we're trying to get things like success rate from the flow, since the number would not add up).

Reply:
Thank you. Btw, most of the BrokerStartEvent and BrokerEndEvent are emit in the same function if you have a look at the code. The getAcquireTokenResult() in the parent class here is called by the BrokerAccountManagerStrategy and BrokerAuthServiceStrategy when extracting the result. We could either move the function back and duplicate the code or add additional try-catch in the calling method. I prefer the current implementation where to emit the result inside getAcquireTokenResult(). However, feel free to refactor the code in the future. :)